docs: improve Multiple Disk Support (#1146)
longhorn/website: content/docs/1.10.0/nodes-and-volumes/nodes/multidisk.md

1. Choose a Disk: Select the physical or virtual disk you want to use for Longhorn storage and format it with a filesystem (e.g., ext4, xfs).
2. Mount the Disk: Mount the disk to a directory on the host, such as `/mnt/example-disk`. Ensure the directory is accessible and properly configured.

Then, you can add the disk using either the Longhorn UI or the `kubectl` command-line tool. Choose the method that suits your preference.

- **Using the Longhorn UI**

  1. Head to the `Node` tab, select one of the nodes, and select `Edit Disks` in the dropdown menu.
  2. Add the path of the mounted disk into the disk list of the node.

- **Using `kubectl` Command**

  1. Execute `kubectl edit node.longhorn.io <node-name>` to edit the node resource.
  2. Add the disk path to `spec.disks` in the node resource. For example:

    ```yaml
    ...
    spec:
      ...
      disks:
        ...
        example-disk:
          allowScheduling: true
          diskDriver: ""
          diskType: filesystem
          evictionRequested: false
          path: /mnt/example-disk
          storageReserved: 0
          tags: []
    ...
    ```

  3. Save and exit the editor.

> Once the disk is added:
> 
> - Longhorn automatically detects the disk’s storage details, such as total and available space.
> - If the disk is suitable for storing volume data, Longhorn will start scheduling volumes to it.
> 
> Optionally, in the `Space Reserved` field, enter the amount of disk space (in bytes) to reserve for other purposes, preventing Longhorn from using that portion.
> 
> To maintain node stability when compute resources like memory or disk space are low, the `kubelet` requires reserved space. Exhaustion of these incompressible resources can lead to node instability. To prevent kubelet `Disk Pressure` issue after scheduling multiple volumes, Longhorn reserves `30%` of the root disk space (`/var/lib/longhorn`) by default. This value is set in the `storage-reserved-percentage-for-default-disk` setting to ensure node stability.
> 
> 
> > **Notice**:
> >
> > 1. You cannot add a disk path that is already in use by another disk.
> >
> > 2. Since Longhorn uses filesystem ID to detect duplicate mounts of the same filesystem, you cannot add a disk that has the same filesystem ID as an existing disk on the same node. See more details at https://github.com/longhorn/longhorn/issues/2477

Reviewer: ```suggestion
Once a disk is added:

- Longhorn automatically detects the storage details of the disk, such as total and available capacity.
- If the disk is suitable for storing volume data, Longhorn begins scheduling volumes to it.

Optionally, you can use the `Space Reserved` field to reserve a portion of disk space (in bytes) for other purposes. This reserved space will not be used by Longhorn for volume data.

To maintain node stability when compute resources (for example memory or disk) are under pressure, the `kubelet` requires some space to remain free. If these critical resources are exhausted, it can lead to node instability.

Longhorn **reserves 30% of the root disk space (`/var/lib/longhorn`) by default** to prevent issues like `DiskPressure` conditions from the `kubelet`, particularly after scheduling multiple volumes. This behaviour is controlled by the `storage-reserved-percentage-for-default-disk` setting.

> **Notice**:
>
> 1. You cannot add a disk path that is already in use by another Longhorn disk.
> 2. Longhorn uses the filesystem ID to detect duplicate mounts. Therefore, you cannot add a disk with the **same filesystem ID** as another disk on the same node.  
>    See: [Issue #2477](https://github.com/longhorn/longhorn/issues/2477)
```

Longhorn supports using more than one disk on the nodes to store the volume data.

By default, Longhorn stores volume data in the `/var/lib/longhorn` directory on the host. If you want to use a different disk for storage, you can add a new disk and disable scheduling for the default directory. This gives you flexibility to manage storage based on your needs.

## Add a Disk

Before adding a disk to Longhorn, you need to mount it to a directory on the host of the Longhorn node.

> 1. Choose a Disk: Select the physical or virtual disk you want to use for Longhorn storage and format it with an extent-based filesystem (e.g., ext4, xfs).
> 2. Mount the Disk: Mount the disk to a directory on the host, such as `/mnt/example-disk`. Ensure the directory is accessible and properly configured.

Reviewer 2: ```suggestion
1. **Choose a Disk:** Select the physical or virtual disk you want to use for Longhorn storage and format it with an extent-based filesystem (e.g., ext4, xfs).
2. **Mount the Disk:** Mount the disk to a directory on the host, such as `/mnt/example-disk`. Ensure the directory is accessible and properly configured.
```

> Then, you can add the disk using either the Longhorn UI or the `kubectl` command-line tool. Choose the method that suits your preference.

Reviewer 2: ```suggestion
After the disk is mounted, you can add it to Longhorn using either the  UI or the `kubectl` command-line tool.
```

- **Using the Longhorn UI**

>   1. Head to the `Node` tab, select one of the nodes, and select `Edit Disks` in the dropdown menu.

Reviewer 2: ```suggestion
  1. Go to the **Node** tab, select a node, and choose **Edit Disks** from the dropdown menu.
```

>   2. Add the path of the mounted disk into the disk list of the node.

Reviewer 2: ```suggestion
  2. Add the mount path of the disk to the disk list.
```

- **Using `kubectl` Command**

>   1. Execute `kubectl edit node.longhorn.io <node-name>` to edit the node resource.

Reviewer 2: ```suggestion
  1. Run `kubectl edit node.longhorn.io <node-name>` to edit the Longhorn node resource.
```

>   2. Add the disk path to `spec.disks` in the node resource. For example:

Reviewer 2: ```suggestion
  2. Add the disk path to `spec.disks`. For example:
```

    ```yaml
    ...
    spec:
      ...
      disks:
        ...
        example-disk:
          allowScheduling: true
          diskDriver: ""
          diskType: filesystem
          evictionRequested: false
          path: /mnt/example-disk
          storageReserved: 0
          tags: []
    ...
    ```

  3. Save and exit the editor.

Once a disk is added:

- Longhorn automatically detects the storage details of the disk, such as maximum and available capacity.
- If the disk is suitable for storing volume data, Longhorn begins scheduling volumes to it.

> Optionally, you can use the `Space Reserved` field in the UI or `spec.disks.<disk-name>.storageReserved` to reserve a portion of disk space (in bytes) for other purposes. This reserved space will not be used by Longhorn for volume data.

Reviewer 2: Consider having this to `### Rook Disk Reservation`.

Author: Good idea. Added, thank you.

To maintain node stability when compute resources (for example memory or disk) are under pressure, the `kubelet` requires some space to remain free. If these critical resources are exhausted, it can lead to node instability.

Longhorn **reserves 30% of the root disk space (`/var/lib/longhorn`) by default** to prevent issues like `DiskPressure` conditions from the `kubelet`, particularly after scheduling multiple volumes. This behaviour is controlled by the `storage-reserved-percentage-for-default-disk` setting.

> **Notice**:
>
> 1. You cannot add a disk path that is already in use by another Longhorn disk.
> 2. Longhorn uses the filesystem ID to detect duplicate mounts. Therefore, you cannot add a disk with the **same filesystem ID** as another disk on the same node.  
>    See: [Issue #2477](https://github.com/longhorn/longhorn/issues/2477)

### Use an Alternative Path for a Disk on the Node

> If you prefer not to use the original mount path of a disk on the node, you can create an alternative (alias) path using `mount --bind` and configure Longhorn to use it. Note that using a symbolic link (`ln -s`) will not work because it won't be correctly recognized inside the pod.

Reviewer 2: ```suggestion
If you prefer to use a different path for a disk (rather than the original mount point), you can use `mount --bind` to create an alternative path. Do **not** use symbolic link (`lh -s`), as these are not properly resolved inside Longhorn pods.
```

> Longhorn identifies disks by their paths, so you must ensure the alternative path is properly mounted after a node reboot, for example, by adding it to `fstab`.

Reviewer 2: ```suggestion
Make sure the alternative path is remounted after a node reboot—for example, by adding it to `/etc/fstab`.
```

## Remove a Disk

Nodes and disks can be excluded from future scheduling. Note that any storage already scheduled on the node will not be automatically released when scheduling is disabled for the node.

> To remove a disk, the following conditions must be met:
> 
> - Scheduling for the disk must be disabled.
> - There must be no replicas or backing images remaining on the disk, including replicas in an error state. For instructions on how to evict replicas from disabled disks, see [Select Disks or Nodes for Eviction](../disks-or-nodes-eviction/#select-disks-or-nodes-for-eviction).

Reviewer 2: ```suggestion
To remove a disk:

- Disable scheduling for the disk.
- Ensure there are **no replicas or backing images** left on the disk, including any in an error state. For instructions on how to evict replicas from disabled disks, see [Select Disks or Nodes for Eviction](../disks-or-nodes-eviction/#select-disks-or-nodes-for-eviction).
```

> Once these conditions are satisfied, you can proceed to remove the disk.

Reviewer 2: ```suggestion
Once the disk is empty and scheduling is disabled, you can safely remove it from the node configuration.
```

## Configuration

There are two global settings affect the scheduling of the volume.

- `StorageOverProvisioningPercentage` defines the maximum total storage that can be **scheduled** on a disk, relative to its usable capacity. The formula is:

    ```bash
    ScheduledStorage / (MaximumStorage - ReservedStorage)
    ```

    The default is `100` (%).

>     For a 200 GiB disk with 50 GiB reserved for the system, the usable capacity is 150 GiB. With the default setting, Longhorn can schedule up to 150 GiB of volume data on this disk.

Reviewer 2: ```suggestion
		On a 200 GiB disk with 50 GiB reserved, Longhorn sees 150 GiB of usable space. With the default setting, it can schedule up to 150 GiB of volume data.
```

>     Since workloads often don’t fully utilise their allocated volumes, and Longhorn stores volume data as sparse files, it is typically safe to increase this value to improve disk utilisation.

Reviewer 2: ```suggestion
    Since workloads typically don’t consume the entire allocated volume size, and Longhorn uses sparse files to store data, increasing this setting is generally safe and can help optimize disk utilization.
```

- `StorageMinimalAvailablePercentage` specifies the minimum percentage of free space that must remain on a disk in order to schedule new replicas. The formula is:

    ```bash
    AvailableStorage / MaximumStorage
    ```

    The default is `25` (%).

>     For example, on a 200 GiB disk with 50 GiB reserved, Longhorn will not schedule new replicas if the available space falls below 37.5 GiB (25% of 150 GiB). A volume also will not be scheduled if its requested size would reduce available storage below this threshold.

Reviewer 2: ```suggestion
    For a 200 GiB disk with 5 GiB reserved, Longhorn will stop scheduling new replicas if available space falls below 37.5 GiB  (25% of 150 GiB). A new volume also will not be scheduled if its size would push available space below that limit.
```

>     This setting helps prevent disks from becoming overly full, which could lead to scheduling failures or volume operation issues.

Reviewer 2: ```suggestion
    This setting helps prevent disks from becoming too full, which could lead to scheduling failures or volume operation issues.
```

> **Warning**:
> 
> Currently, Longhorn cannot fully enforce the `StorageMinimalAvailablePercentage` limit in all scenarios because:
>

> > - Longhorn volumes may use more space than their specified size since snapshots store previous states of the volume.

Reviewer 2: ```suggestion
> - Longhorn volumes may use more space than their requested size, especially when snapshots are taken.
```

Once the disk is empty and scheduling is disabled, you can safely remove it from the node configuration.

## Configuration

There are two global settings affect the scheduling of the volume.

- `StorageOverProvisioningPercentage` defines the maximum total storage that can be **scheduled** on a disk, relative to its usable capacity. The formula is:

    ```bash
    ScheduledStorage / (MaximumStorage - ReservedStorage)
    ```

    The default is `100` (%).

	On a 200 GiB disk with 50 GiB reserved, Longhorn sees 150 GiB of usable space. With the default setting, it can schedule up to 150 GiB of volume data.

    Since workloads typically don’t consume the entire allocated volume size, and Longhorn uses sparse files to store data, increasing this setting is generally safe and can help optimize disk utilization.

- `StorageMinimalAvailablePercentage` specifies the minimum percentage of free space that must remain on a disk in order to schedule new replicas. The formula is:

    ```bash
    AvailableStorage / MaximumStorage
    ```

    The default is `25` (%).

>     For a 200 GiB disk with 5 GiB reserved, Longhorn will stop scheduling new replicas if available space falls below 37.5 GiB  (25% of 150 GiB). A new volume also will not be scheduled if its size would push available space below that limit.

Reviewer 2: ```suggestion
    For a 200 GiB disk with 50 GiB reserved, Longhorn will stop scheduling new replicas if available space falls below 37.5 GiB  (25% of 150 GiB). A new volume also will not be scheduled if its size would push available space below that limit.
```

Longhorn will identify the disk using the path, so the users need to make sure the alternative path are correctly mounted when the node reboots, e.g. by adding it to `fstab`.

## Remove a Disk

Nodes and disks can be excluded from future scheduling. Notice that any scheduled storage space won't be released automatically if the scheduling was disabled for the node.

In order to remove a disk, two conditions need to be met:

- The scheduling for the disk must be disabled
- There is no existing replica using the disk, including any replicas in an error state. About how to evict replicas from disabled disks, refer to [Select Disks or Nodes for Eviction](../disks-or-nodes-eviction/#select-disks-or-nodes-for-eviction)

Once those two conditions are met, you should be allowed to remove the disk.

## Configuration

There are two global settings affect the scheduling of the volume.

> - `StorageOverProvisioningPercentage` sets the upper limit of storage that can be scheduled on a disk. It is calculated as:

Author: @sushant-suse 
`StorageOverProvisioningPercentage` is missing?
